digdag-ui/console.jsx

```diff
@@ -1920,6 +1950,313 @@ class WorkflowRevisionPage extends React.Component {
   }
 }
 
+class FileEditor extends React.Component {
+  editor: ?CodeEditor
+
+  props:{
+    file: ?TarEntry;
+    projectArchive: ?ProjectArchive;
+  }
+
+  state:{
+    name: string;
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      name: props.file ? props.file.name : 'new.dig'
+    }
+  }
+
+  getName () {
+    return this.state.name
+  }
+
+  getFileContents () {
+    var text
+    if (this.editor) {
+      text = this.editor.getValue()
+    } else {
+      text = this.props.file ? fileString(this.props.file.name, this.props.projectArchive) : ''
+    }
+
+    const buffer = new Buffer(text)
+    const ab = new ArrayBuffer(buffer.length);
+    const view = new Uint8Array(ab);
+    for (var i = 0; i < buffer.length; ++i) {
```

Comment:
does `view.set(buffer)` works in this case? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/set
I'm not a great fan of for loops or copying over like this